airflow/www/views.py

```diff
@@ -1122,6 +1122,47 @@ def blocked(self, session=None):
             })
         return wwwutils.json_response(payload)
 
+    @expose('/dagrun_failed')
+    @login_required
+    @wwwutils.action_logging
+    @wwwutils.notify_owner
+    def dagrun_failed(self):
```

Comment:
Seems like there is a lot of duplication with dagrun_success, does it make sense to factor out the bulk of the logic?